### PR TITLE
Permutation Example

### PR DESCRIPTION
**Pull request by harrisonbrock**

### CODE REVIEW

File: topics/go/algorithms/strings/permuation/permutation.go [hunk 1]
```diff
@@ -5,7 +5,8 @@ import (
 	"strings"
 )
 
-// IsPermutation check if two strings are permutations.
+// IsPermutation check if two
+// strings are permutations.
 func IsPermutation(str1, str2 string) bool {
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/strings/permuation/permutation.go:10]: Sorry, this is short enough to be on 1 line.

File: topics/go/algorithms/strings/permuation/permutation.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+func IsPermutation(str1, str2 string) bool {
+
+	// If the length are not equal they cannot be permutation.
+	if len(str1) != len(str2) {
+		return false
+	}
+
+	// Create a slice for each input string that
+	//is split on every character of the string.
+	s1 := strings.Split(str1, "")
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/strings/permuation/permutation.go:19]: Look at the comment on line 12, that is a good length before the break.

// <space> is split

File: topics/go/algorithms/strings/permuation/permutation.go [hunk 3]
```diff
@@ -1,27 +1,25 @@
 package strings
 
 import (
+	strings "github.com/ardanlabs/gotraining/topics/go/algorithms/strings/types"
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/strings/permuation/permutation.go:4]: Don't do this. Keep everything together in this single file. It doesn't matter if we need it again.

File: topics/go/algorithms/strings/permuation/permutation.go [hunk 4]
```diff
@@ -1,27 +1,25 @@
...
-	s1 := strings.Split(str1, "")
-	s2 := strings.Split(str2, "")
+	// Create a rune for each input string.
+	s1 := []rune(str1)
+	s2 := []rune(str2)
 
-	// Sort the the two slices.
-	sort.Strings(s1)
-	sort.Strings(s2)
+	// Sort the the two runes
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/strings/permuation/permutation.go:20]: Check the comment.

File: topics/go/algorithms/strings/permuation/permutation.go (hunk 3 above)
Comment (harrisonbrock) [reply topics/go/algorithms/strings/permuation/permutation.go:4]: Ok, Thanks.

File: topics/go/algorithms/strings/permuation/permutation.go [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package strings
+
+import (
+	"sort"
+)
+
+type RuneSlice []rune
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/strings/permuation/permutation.go:7]: Check you golint tooling is active. This exported type would need a coment.3

File: topics/go/algorithms/strings/permuation/permutation.go [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+// For sorting an RuneSlice
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/strings/permuation/permutation.go:9]: Not sure what happens here with comments and goling. Missing a period.

File: topics/go/algorithms/strings/permuation/permutation.go [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+// IsPermutation check if two strings are permutations.
+func IsPermutation(str1, str2 string) bool {
+
+	// If the length are not equal they cannot be permutation.
+	if len(str1) != len(str2) {
+		return false
+	}
+
+	// Create a rune for each input string.
+	s1 := []rune(str1)
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/strings/permuation/permutation.go:23]: // Convert each string into its collection of runes.

File: topics/go/algorithms/strings/permuation/permutation.go [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+	s2 := []rune(str2)
+
+	// Sort the the two runes.
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/strings/permuation/permutation.go:26]: // Sort each collection of runes.

File: topics/go/algorithms/strings/permuation/permutation.go [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+	sort.Sort(RuneSlice(s1))
+	sort.Sort(RuneSlice(s2))
+
+	return string(s1) == string(s2)
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/strings/permuation/permutation.go:30]: // Convert the collection of runes back to a string and compare.

File: topics/go/algorithms/strings/permuation/permutation_test.go [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+import (
+	strings "github.com/ardanlabs/gotraining/topics/go/algorithms/strings/permuation"
+	"testing"
+)
+
+const succeed = "\u2713"
+const failed = "\u2717"
+
+func TestIsPermutation(t *testing.T) {
+
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/strings/permuation/permutation_test.go:12]: Have we been leaving this line space with the other tests?

**Pull request merged**